### PR TITLE
Don't include failed item rewrites in missed spans

### PR DESCRIPTION
**Pull request by marcusklaas**
Fix https://github.com/rust-lang-nursery/rustfmt/issues/661.

### CODE REVIEW

File: tests/source/structs.rs [hunk 1]
```diff
@@ -32,7 +32,7 @@ struct Qux<'a,
 (
     AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA, // Comment
     BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB,
-    #[AnAttr]    
+    #[AnAttr]
```
Comment (nrc) [inline tests/source/structs.rs]: Isn't this a valid test? We should whitespace from attributes.

**Comment by nrc** [comment]
r+ other than the test question

**Comment by marcusklaas** [comment]
It's a valid test, but we were never rewriting the struct. Not sure why we fail here. The whitespace was stripped as part of the missing span routines, but that also did things like de-indenting comments. I guess it's a trade-off. We should look into why we can't format this particular struct, though.

**Comment by nrc** [comment]
OK, I didn't realise we just didn't reformat - I'll file an issue for it

**Pull request merged**